components/core/src/clp_s/search/kql/kql.cpp

```diff
@@ -76,6 +79,72 @@ class ParseTreeVisitor : public KqlBaseVisitor {
         }
     }
 
+    static std::shared_ptr<Literal> create_timestamp_literal(
+            KqlParser::Timestamp_expressionContext const& ctx
+    ) {
+        if (nullptr == ctx.timestamp) {
+            return nullptr;
+        }
+
+        auto const timestamp_str{ctx.timestamp->getText()};
+        std::string generated_pattern;
+        if (ctx.pattern) {
+            auto const pattern_str{ctx.pattern->getText()};
+            auto const pattern_result{timestamp_parser::TimestampPattern::create(pattern_str)};
+            if (pattern_result.has_error()) {
+                SPDLOG_ERROR(
+                        "Invalid timestamp pattern {} - {}.",
+                        pattern_str,
+                        pattern_result.error().message()
+                );
+                return nullptr;
+            }
+
+            auto const timestamp_result{timestamp_parser::parse_timestamp(
+                    timestamp_str,
+                    pattern_result.value(),
+                    true,
+                    generated_pattern
+            )};
+            if (timestamp_result.has_error()) {
+                SPDLOG_ERROR(
+                        "Failed to parse timestamp {} using pattern {} - {}",
+                        timestamp_str,
+                        pattern_str,
+                        timestamp_result.error().message()
+                );
+                return nullptr;
+            }
+            return DateLiteral::create(timestamp_result.value().first);
+        }
+
+        auto const quoted_patterns_result{
+                timestamp_parser::get_all_default_quoted_timestamp_patterns()
+        };
+        if (quoted_patterns_result.has_error()) {
+            SPDLOG_ERROR(
+                    "Unexpected error while trying to load default timestamp patterns - {}",
+                    quoted_patterns_result.error().message()
+            );
+            return nullptr;
+        }
+
+        auto const optional_timestamp{timestamp_parser::search_known_timestamp_patterns(
+                timestamp_str,
+                quoted_patterns_result.value(),
+                true,
+                generated_pattern
+        )};
+        if (false == optional_timestamp.has_value()) {
+            SPDLOG_ERROR(
+                    "Failed to parse timestamp {} using default timestamp patterns.",
+                    timestamp_str
+            );
+            return nullptr;
+        }
+        return DateLiteral::create(optional_timestamp.value().first);
```

Comment:
Very nit: we can dereference it:

```suggestion
        return DateLiteral::create(optional_timestamp->first);
```